source/fundamentals/crud/write-operations/bulk-write.txt

```diff
@@ -0,0 +1,550 @@
+.. _csharp-bulk-write:
+
+=====================
+Bulk Write Operations
+=====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: insert, update, replace, code example
+
+Overview
+--------
+
+This guide shows you how to use the {+driver-short+} to perform **bulk write operations**
+that include multiple write operations in a single database call.
+
+Consider a scenario in which you want to insert a document into a collection,
+update multiple other documents, then delete a document. If you use
+individual methods, each operation requires its own database call. If you use
+a bulk write operation, you can improve efficiency by performing multiple write
+operations in a single database call.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<csharp-quickstart>` tutorial.
+
+Define the Write Operations
+---------------------------
+
+For each write operation you want to perform, create an instance of one of
+the following ``BulkWriteModel`` classes:
+
+- ``BulkWriteInsertOneModel<TDocument>``
+- ``BulkWriteUpdateOneModel<TDocument>``
+- ``BulkWriteUpdateManyModel<TDocument>``
+- ``BulkWriteReplaceOneModel<TDocument>``
+- ``BulkWriteDeleteOneModel<TDocument>``
+- ``BulkWriteDeleteManyModel<TDocument>``
+
+The following sections show how to create and use instances of the preceding classes
+to perform the corresponding write operation in a bulk write.
+
+.. tip:: Bulk Write Operations with POCOs
+
+    The examples in this guide use the ``BsonDocument`` type for the ``TDocument`` type
+    in all generic classes. You can also use a Plain Old CLR Object (POCO) for these
```

Comment:
Q: what does "in all generic classes" mean in this context?

Reply:
The classes on this page that use `TDocument` as a type parameter, like all of the BulkWriteModels. I'd expect this term to be familiar to C# devs using this page